src/use_case/take_quiz/TakeQuizInteractor.java

```diff
@@ -15,25 +18,90 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 public class TakeQuizInteractor{
-  public ArrayList<Song> generateRandomSongs(){
+
+    // STEP 1: Create a SongFactory object to use in DAO
+    // IMPORTANT NOTE: FIX THIS!!
+    SongFactory songFactory = new SongFactory(); // FIX!!!
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // STEP 2: Use DAO to create Masterlist of Songs from DAO
+    QuizDataAccessObject quizDao = new QuizDataAccessObject(songFactory);
+
+    // IMPORTANT NOTE: below code "try/catch", etc. is to handle the case where Quiz.readcsv throws an exception
+    {
+        try {
+            quizDao.readCSV();  // Setting quizDao.songlist attribute to be the masterlist
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }// Setting quiz's songlist attribute using csv
+    ArrayList<Song> masterlist = quizDao.getSongList();
+
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // STEP 3: Create Quiz Object
+    Quiz quiz = new Quiz();
+    // IMPORTANT NOTE: Below quiz.setNumQuestions method doesn't work unless...
+    // you put {} around it, it's because it returns void.
+    // Find a way to set limit of input to be <= 50.
+    {quiz.setNumQuestions(50);}
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // STEP 4: Helper method for GENERATING 2 RANDOM SONGS FROM MASTERLIST
+    public ArrayList<Song> generate2RandomSongs(){
        // Creating a Random object
         Random rand = new Random();
 
         // Creating an empty ArrayList to store 2 random songs
         ArrayList<Song> finalSongs = new ArrayList<Song>();
 
         // Getting a random list index 1 and 2
-        int ranIndex1 = rand.nextInt(this.songList.size());
-        int ranIndex2 = rand.nextInt(this.songList.size());
-        // If indexes are equal, we choose a new random index
-        while(ranIndex1 == ranIndex2){
-            ranIndex2 = rand.nextInt(this.songList.size());
+        int size = masterlist.size();
+        int ranIndex1 = rand.nextInt(size);
+        int ranIndex2 = rand.nextInt(size);
+
+        Song ranSong1 = masterlist.get(ranIndex1);
+        Song ranSong2 = masterlist.get(ranIndex2);
+
+        // If indexes are equal OR songs have same popularity, we choose a new random index/song
+        while((ranIndex1 == ranIndex2) || (ranSong1.getPopularity() == ranSong2.getPopularity())){
+            ranIndex2 = rand.nextInt(masterlist.size());
```

Comment:
maybe make a helper method for choosing the songs on the off chance that it chooses the same index/popularity again, this way only accounts for it doing that one time